utils/kata-manager.sh

```diff
@@ -265,31 +293,54 @@ Description: Install $kata_project [1] (and optionally $containerd_project [2])
 
 Options:
 
- -c <flavour> : Specify containerd flavour ("lts" | "active" - default: "lts").
-                Find more details on LTS and Active versions of containerd on
-                https://containerd.io/releases/#support-horizon
- -d           : Enable debug for all components.
- -D           : Install Docker server and CLI tooling (takes priority over '-c').
- -f           : Force installation (use with care).
- -h           : Show this help statement.
- -k <version> : Specify Kata Containers version.
- -K <tarball> : Specify local Kata Containers tarball to install (takes priority over '-k').
- -l           : List installed and available versions only, then exit (uses network).
- -N           : Install nerdctl (takes priority over '-c', only implemented for x86_64 and ARM).
- -o           : Only install Kata Containers.
- -r           : Don't cleanup on failure (retain files).
- -t           : Disable self test (don't try to create a container after install).
- -T           : Only run self test (do not install anything).
+ -c <flavour>    : Specify containerd flavour ("lts" | "active" - default: "lts").
+                   Find more details on LTS and Active versions of containerd on
+                   https://containerd.io/releases/#support-horizon
+ -d              : Enable debug for all components.
+ -D              : Install Docker server and CLI tooling (takes priority over '-c').
+ -e              : List short names and details for local hypervisor configuration files (for '-S').
+ -f              : Force installation (use with care).
+ -h              : Show this help statement.
+ -H <hypervisor> : Specify the hypervisor name to use when *INSTALLING* a system.
+ -k <version>    : Specify Kata Containers version.
+ -K <tarball>    : Specify local Kata Containers tarball to install (takes priority over '-k').
+ -l              : List installed and available versions only, then exit (uses network).
+ -L              : List short names and details for official packaged hypervisor configurations (for '-S').
+ -N              : Install nerdctl (takes priority over '-c', only implemented for x86_64 and ARM).
+ -o              : Only install Kata Containers.
+ -r              : Don't cleanup on failure (retain files).
+ -S <hypervisor> : Only change the hypervisor config for an *EXISTING* installation.
+ -t              : Disable self test (don't try to create a container after install).
+ -T              : Only run self test (do not install anything).
 
 Notes:
 
 - The version strings must refer to official GitHub releases for each project.
   If not specified or set to "", install the latest available version.
 
+- The '-L' option requires an installed system.
+
+  > **Note:** For details of each hypervisor, see [3].
+
+- If an invalid hypervisor name is specified with the '-H' option, an
```

Comment:
Personally I do not like the concept of designating any partcular hypervisor as default. I feel all the hypervisors should be treated the same with the user choosing one based on his/her needs.
(While I do understand that configuration.toml points to qemu conf by default, we should limit the usage of default and not designate any one hypervisor as default)

If there is a failure on configuration here, could we just go back to previous configuration instead of default?

Reply:
Hi, @amshinde - I don't disagree with what you've said, but we need some way to designate "the currently configured hypervisor". How about just `configured`?

Reply:
@jodh-intel Yes that wording works for me.

Reply:
@amshinde - Switching everything to `configured` unfortunately results in a less clear approach. For example, how does the user "undo all changes" and revert to the pristine default configured hypervisor?

The current approach is:

```bash
$ kata-manager.sh -S default
```

But that just looks odd if we change the name to `configured` as it just doesn't express the same intent:

```bash
$ kata-manager.sh -S configured
```

That's confusing and ambiguous imho as it doesn't suggest the value specified to the `-S` option means "revert to the original configured hypervisor".

The best I can come up with as a compromise is:


```bash
$ kata-manager.sh -S pristine
```

wdyt?

/cc @fidencio